### PR TITLE
Datastore FE

### PR DESCRIPTION
**Pull request by NaamaHolzer**
Connect the front end to the back end - use 'get' to fetch existing markers from the Datastore and 'post' to store new ones.

### CODE REVIEW

File: capstone-project/src/main/webapp/map/map.component.js [hunk 1]
```diff
@@ -71,12 +72,24 @@ angular.module('map').component('mapComponent', {
             return containerDiv;
         }
 
+        // Sends a marker to the backend for saving.
+        function postMarker(marker) {
```
Comment (NoyFlat) [inline capstone-project/src/main/webapp/map/map.component.js:76]: All other function are anonymous functions, and this one is a regular function definition.
I think it's better if we just choose one way to declare functions and stick to it.

**Review by NoyFlat — CHANGES_REQUESTED** [review]

File: capstone-project/src/main/webapp/map/map.component.js [hunk 2]
```diff
@@ -88,7 +101,10 @@ angular.module('map').component('mapComponent', {
             });
         };
 
-        // Iterate over the markers json and add all of them to the map
-        $http.get('map/markers/markers.json').then((response) => angular.forEach(response.data, addMarkerForDisplay));
+        //Fetches markers from the backend and adds them to the map.
```
Comment (NoyFlat) [inline capstone-project/src/main/webapp/map/map.component.js:104]: Missing a space after //

File: capstone-project/src/main/webapp/map/map.component.js (hunk 1 above)
Comment (orarch) [inline capstone-project/src/main/webapp/map/map.component.js]: There's inconsistency in this file on defining functions, e.g, buildInfoWindowInput is assigned to a variable. 

Let's be consistent if we can, I like it better the way you defined it here.

File: capstone-project/src/main/webapp/map/map.component.js [hunk 3]
```diff
@@ -88,7 +101,10 @@ angular.module('map').component('mapComponent', {
...
+        $http({
```
Comment (orarch) [inline capstone-project/src/main/webapp/map/map.component.js:105]: Why not use the $http.get method? 

https://docs.angularjs.org/api/ng/service/$http#get

File: capstone-project/src/main/webapp/map/map.component.js [hunk 4]
```diff
@@ -71,12 +72,24 @@ angular.module('map').component('mapComponent', {
             return containerDiv;
         }
 
+        // Sends a marker to the backend for saving.
+        function postMarker(marker) {
+
+            const markerJson = JSON.stringify(marker);
+            const data = { marker: markerJson }
+            $http({
```
Comment (orarch) [inline capstone-project/src/main/webapp/map/map.component.js:80]: Why not use the $http.post method? 
https://docs.angularjs.org/api/ng/service/$http#post

**Review by NoyFlat — APPROVED** [review]

**Review by orarch — APPROVED** [review]

**Pull request merged**